gensim/models/word2vec.py

```diff
@@ -1144,9 +1152,43 @@ def __iter__(self):
                         i += self.max_sentence_length
 
 
+def _scan_vocab_worker(stream, progress_queue, max_vocab_size=None, trim_rule=None):
+    """Do an initial scan of all words appearing in stream.
+
+    Note: This function can not be Word2VecVocab's method because
+    of multiprocessing synchronization specifics in Python.
+    """
+    min_reduce = 1
+    vocab = defaultdict(int)
+    checked_string_types = 0
+    sentence_no = -1
+    total_words = 0
+    for sentence_no, sentence in enumerate(stream):
+        if not checked_string_types:
+            if isinstance(sentence, string_types):
+                log_msg = "Each 'sentences' item should be a list of words (usually unicode strings). " \
+                          "First item here is instead plain %s." % type(sentence)
+                progress_queue.put(log_msg)
+
+            checked_string_types += 1
```

Comment:
This seems to be functioning as a boolean in this construct. Any particular reason for defining it as an integer and incrementing?

Reply:
I've just copied this logic from an old version of the code (I don't know why it's not bool, but I could fix it if you insist)